### PR TITLE
i#3544 RV64: Add the missing architecture-related code to drcachesim

### PR DESCRIPTION
**Pull request by chenhy0106**
Fix the issue in drcachesim where missing architecture-related code leads to incorrect counting results.

Issue: #3544

### CODE REVIEW

File: clients/drcachesim/tracer/instru_online.cpp [hunk 1]
```diff
@@ -332,6 +332,23 @@ online_instru_t::insert_save_type_and_size(void *drcontext, instrlist_t *ilist,
...
+                XINST_CREATE_load_int(drcontext, opnd_create_reg(scratch),
+                                      OPND_CREATE_INT(size)));
+        /* SLLI scratch, scratch, 16 */
+        MINSERT(ilist, where,
+                INSTR_CREATE_slli(drcontext, opnd_create_reg(scratch),
+                                  opnd_create_reg(scratch), OPND_CREATE_INT8(16)));
+        /* ORI scratch, scratch, #type */
+        MINSERT(ilist, where,
+                INSTR_CREATE_ori(drcontext, opnd_create_reg(scratch),
+                                 opnd_create_reg(scratch), OPND_CREATE_INT(type)));
```
Comment (ksco) [inline clients/drcachesim/tracer/instru_online.cpp:348]: So we need to store `type | (size << 16)` to some memory address, can we use `insert_mov_immed_ptrsz()` to load the immediate?

File: clients/drcachesim/tracer/tracer.cpp [hunk 2]
```diff
@@ -684,6 +684,11 @@ insert_conditional_skip(void *drcontext, instrlist_t *ilist, instr_t *where,
     MINSERT(ilist, where,
             INSTR_CREATE_cbz(drcontext, opnd_create_instr(skip_label),
                              opnd_create_reg(reg_skip_if_zero)));
+#elif defined(RISCV64)
+    MINSERT(ilist, where,
+            INSTR_CREATE_beq(drcontext, opnd_create_instr(skip_label),
+                             opnd_create_reg(reg_skip_if_zero),
+                             opnd_create_reg(DR_REG_X0)));
```
Comment (ksco) [inline clients/drcachesim/tracer/tracer.cpp:691]: s/DR_REG_X0/DR_REG_ZERO/ to be more readable?

**Review by ksco — APPROVED** [review]

File: clients/drcachesim/tracer/instru_online.cpp [hunk 3]
```diff
@@ -332,6 +332,15 @@ online_instru_t::insert_save_type_and_size(void *drcontext, instrlist_t *ilist,
         MINSERT(ilist, where,
                 XINST_CREATE_store(drcontext, OPND_CREATE_MEM32(base, disp),
                                    opnd_create_reg(scratch)));
+#elif defined(RISCV64)
```
Comment (derekbruening) [inline clients/drcachesim/tracer/instru_online.cpp:335]: Not related to this PR: there is someone asking about building DR for RISC-V on the users list: https://groups.google.com/g/dynamorio-users/c/7LYfNgD8OEA

Hoping you or someone in your group could help answer it.

**Review by derekbruening — APPROVED** [review]

File: clients/drcachesim/tracer/instru_online.cpp (hunk 3 above)
Comment (ksco) [reply clients/drcachesim/tracer/instru_online.cpp:335]: Ack.

**Pull request merged**